src/eterna/ui/AnnotationDialog.ts

```diff
@@ -59,6 +59,23 @@ export default class AnnotationDialog extends Dialog<AnnotationData> {
 
     protected added(): void {
         super.added();
+        this.renderDialog(true);
+    }
+
+    public renderDialog(focusTitle: boolean): void {
+        const prevTitle = this._titleField?.input.text || '';
+        const prevBases = this._basesField?.input.text || '';
+
+        // Clear old objects
+        this.removeAllObjects();
+        this.container.children.forEach((child: DisplayObject) => {
+            child.destroy();
+        });
+
+        if (this._modal) {
+            this.setupModalBackground();
+        }
```

Comment:
Could you explain what necessitated this?

Reply:
When there is an update to information that the dialog consumes, it prompts a rerender. I'd opted for refreshing the class instance information vs creating a new class instance. In the case where we have the "edit" version of the AnnotationDialog, we'd lose the modal background when we clear the child objects. This effectively  brings it back

Reply:
Ah, it's an effect of removing all child objects - that's another rationale for keeping objects instantiated